osu.Game.Rulesets.Osu/Mods/OsuMod.cs

```diff
@@ -26,10 +27,43 @@ public class OsuModEasy : ModEasy
 
     }
 
-    public class OsuModHidden : ModHidden
+    public class OsuModHidden : ModHidden, IApplicableToDrawableHitObjects

```

Comment:
Am not really a fan of directly referencing the drawable representations of hit objects and changing their properties. I think better is for the drawable representations to handle their own fadeouts.

How about changing this to inherit from IApplicableToHitObject, and setting a field like `PreEmptFadeOut` on the hitobject?

Reply:
So the idea is to move all new fields I exposed to the hitobject instead? I'll try once at home.

Reply:
Mh. If I move all of them they'd have to be in `OsuHitObject` as I'd have to cast for some fields otherwise. That doesn't sound desirable.
But if I move them all to `OsuHitObject` a field named `ShowApproachCircle` wouldn't make much sense (or one of the other 3).
If I merge them together under a generic name it isn't really specific what that field would do.

So I'd like to keep it as it is. What do you think?

Reply:
@smoogipoo I summon thee for an opinion. So I can continue over the weekend if needed.

Reply:
I don't understand the reason for why it doesn't make sense to put it in the hitobject?

Reply:
This was mostly about those 4 flags that disable or hide something. Some changes are quite specific to the respective type of HitObject. Take hiding the ApproachCircles as an example.
I wouldn't be able to move that one into HitCircle as I'd have to cast the HitObject each time to see if it actually is a HitCircle. Having it in HitObject would work but Spinners and Sliders (except for their InitialCircle) don't have ApproachCircles.
It wouldn't feel right to have them there. Unless I'm overthinking this...